lib/ae_mdw/collection.ex

```diff
@@ -69,6 +69,22 @@ defmodule AeMdw.Collection do
     end
   end
 
+  @doc """
+  Streams forward a table seeking the iterator to a boundary start key.
+  """
+  @spec stream(table(), key()) :: Enumerable.t()
+  def stream(table, boundary_start_key) do
+    Stream.unfold(
+      boundary_start_key,
+      fn key ->
+        case Database.next_key(table, key) do
+          {:ok, next_key} -> {next_key, next_key}
+          :none -> nil
+        end
+      end
+    )
```

Comment:
If the first key is the same as the boundary start key this function will not return it. Perhaps we can reuse the `unfold_stream` function which is more general?

Reply:
Exactly this is to replace the `:mnesia.select` on a simpler way when we don't have the first key and care only for a pattern like here:
```
    |> Collection.stream({height_pos, kind, <<>>, -1})
    |> Stream.take_while(fn
      {^height_pos, ^kind, _target, _ref} -> true
      _other_height_kind -> false
    end)
```
I will fetch the first key though like the boundary function.